examples/recording_service/pluggable_storage/c/FileStorageWriter.c

```diff
@@ -194,16 +194,26 @@ int FileStorageWriter_connect(void *storage_writer_data)
             (struct FileStorageWriter *) storage_writer_data;
     int64_t current_time = -1;
 
+#ifdef RTI_WIN32
+    fopen_s(&writer->file.file, writer->file.file_name, "w");
```

Comment:
While I **hate** having to use different code when there is a standard library function, `fopen_s` is supposed to be better, more secure, less exploitable. In general, all the `_s` functions on Windows are better options than their standard counterparts.
Side question: in Osapi, for the functions that we define wrappers for, are we using this version of the functions on Windows?

Reply:
no, our file open function is a simple wrapper of fopen. Due to that, we will face the same warning issue.The strncat and strncpy function use their own implementation